hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/s2sjoin/SqlSTSInnerEquiJoinFaultToleranceStressTest.java

```diff
@@ -0,0 +1,260 @@
+                "SELECT * FROM TABLE(IMPOSE_ORDER(TABLE " + topicName + " , DESCRIPTOR(__key), 5))");
+
+        query = "CREATE JOB job OPTIONS ('processingGuarantee'='" + processingGuarantee + "') AS " +
+                " SINK INTO " + resultMapName +
+                " SELECT s1.__key, s2.this FROM s1 JOIN s2 ON s1.__key = s2.__key";
+
+        Thread.sleep(120_000L);
+    }
+
+    @Override
+    @After
+    public void tearDown() {
+        try {
+            Thread.sleep(2000L);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void stressTest() {
+        JobRestarter jobRestarter = new JobRestarter(instance().getJet());
+        jobRestarter.start();
+
+        assertQueryFinished(query, resultMap, EVENTS_TO_PROCESS);
+        assertEquals(EVENTS_TO_PROCESS, resultMap.size());
+        jobRestarter.finish();
+
+        for (int i = 1; i <= EVENTS_TO_PROCESS; ++i) {
+            assertEquals("value-" + i, resultMap.remove(i));
+        }
+        assertEquals(0, resultMap.size());
+
+        try {
+            kafkaFeedThread.join();
+            jobRestarter.join();
+        } catch (AssertionError e) {
+            Assert.fail(e.getMessage());
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    class JobRestarter extends Thread {
+        private final JetService jetService;
+        private volatile boolean finish;
+
+        JobRestarter(JetService jetService) {
+            this.jetService = jetService;
+        }
+
+        @Override
+        public void run() {
+            for (; ; ) {
+                try {
+                    if (finish) {
+                        return;
+                    }
+                    Thread.sleep(500L);
+                    assert jetService.getJobs().size() == 1 : "Jobs active " + jetService.getJobs().size();
+                    Job job = jetService.getJobs().get(0);
+                    assertNotNull(job);
+                    assertJobStatusEventually(job, RUNNING);
+//                    restartGraceful ? job.restart() : job.;
+                    job.restart();
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+
+        public void finish() {
+            this.finish = true;
+        }
+    }
+
+    private static String createRandomTopic() {
+        String topicName = randomName();
+        kafkaTestSupport.createTopic(topicName, INITIAL_PARTITION_COUNT);
+        return topicName;
+    }
+
+    private static void createTopicData(SqlService sqlService, String topicName) {
+        int itemsSank = 0;
+        for (int sink = 1; sink <= SINK_ATTEMPTS; sink++) {
+            StringBuilder queryBuilder = new StringBuilder("INSERT INTO " + topicName + " VALUES ");
+            for (int i = 1; i <= EVENTS_PER_SINK; ++i) {
```

Comment:
Instead of this huge query you can use `kafkaTestSupport.produce()`. SQL might be better after datalinks are implemented, when rows can be inserted one by one with a parametric query with a good performance.